streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -740,8 +740,9 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
         consumerProps.put(NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
         consumerProps.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, StreamPartitionAssignor.class.getName());
         consumerProps.put(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG, getLong(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG));
-
         consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
+        final AdminClientConfig config = new AdminClientConfig(getClientPropsWithPrefix(ADMIN_CLIENT_PREFIX, AdminClientConfig.configNames()));
+        consumerProps.put(adminClientPrefix(AdminClientConfig.RETRIES_CONFIG), config.getInt(AdminClientConfig.RETRIES_CONFIG));
```

Comment:
I don't think so. We could have a global `retries` and an `admin.retries` config and we need the detour via `AdminClientConfig` to pick the correct one. Doing it "manually" here and check if `admin.retries` exist and use `retires` if it does not exist has the issue that `retires` might be missing, too. For this case, we need to use `AdminClientConfig` default value for `retries`, but this is only available via `AdminClientConfig`.

Or is there a simpler way?

Reply:
Yeah I was thinking about doing that "manually" to first check if the `admin.retries` exist, and then if `retries` exist, and last use default value. I see your point that this may not be simpler than what you wrote here.